### PR TITLE
remove deprecated api

### PR DESCRIPTION
**Pull request by jmecosta**
#46  replace profile deprecated apis

### CODE REVIEW

File: sonar-fsharp-plugin/src/main/java/org/sonar/plugins/fsharp/FSharpPlugin.java [hunk 1]
```diff
@@ -55,7 +55,7 @@
   public static final String FILE_SUFFIXES_KEY = "sonar.fs.file.suffixes";
   public static final String FILE_SUFFIXES_DEFVALUE = ".fs,.fsx,.fsi";
 
-  public static final String FSHARP_WAY_PROFILE = "Sonar way";
+  public static final String FSHARP_WAY_PROFILE = "F# Rules";
```
Comment (milbrandt) [inline sonar-fsharp-plugin/src/main/java/org/sonar/plugins/fsharp/FSharpPlugin.java]: Should the Name indicate, that these are the Plugins default rule set? eg. "F# default rules"

File: sonar-fsharp-plugin/src/main/java/org/sonar/plugins/fsharp/FSharpSonarWayProfile.java [hunk 2]
```diff
@@ -32,63 +32,61 @@
...
-import org.sonar.api.profiles.RulesProfile;
-import org.sonar.api.rules.Rule;
-import org.sonar.api.utils.ValidationMessages;
+import org.sonar.api.server.profile.BuiltInQualityProfilesDefinition;
 
-// based on plugins from from https://github.com/SonarSource
-public class FSharpSonarWayProfile extends ProfileDefinition {
+public class FSharpSonarWayProfile implements BuiltInQualityProfilesDefinition {
+
+  private static void activateRule(NewBuiltInQualityProfile profile, String ruleKey, String severity) {
```
Comment (milbrandt) [inline sonar-fsharp-plugin/src/main/java/org/sonar/plugins/fsharp/FSharpSonarWayProfile.java]: why `static`? Called from the instance method `define`, no access to static data needed.

**Comment by milbrandt** [comment]
In `org.sonar.plugins.fsharp.FSharpSensor` there are several other deprecated API used:
- CoreMetrics.FILE_COMPLEXITY_DISTRIBUTION
- CoreMetrics.FUNCTION_COMPLEXITY_DISTRIBUTION
- org.sonar.api.batch.fs.FileSystem.files

**Review by milbrandt — CHANGES_REQUESTED** [review]

**Comment by jmecosta** [comment]
> In `org.sonar.plugins.fsharp.FSharpSensor` there are several other deprecated API used:
> 
> * CoreMetrics.FILE_COMPLEXITY_DISTRIBUTION
> * CoreMetrics.FUNCTION_COMPLEXITY_DISTRIBUTION
> * org.sonar.api.batch.fs.FileSystem.files

Can habdle that in a new pr.

File: sonar-fsharp-plugin/src/main/java/org/sonar/plugins/fsharp/FSharpPlugin.java (hunk 1 above)
Comment (jmecosta) [reply sonar-fsharp-plugin/src/main/java/org/sonar/plugins/fsharp/FSharpPlugin.java]: Sure, can check what they have as default in last versioslns

File: sonar-fsharp-plugin/src/main/java/org/sonar/plugins/fsharp/FSharpSonarWayProfile.java (hunk 2 above)
Comment (jmecosta) [reply sonar-fsharp-plugin/src/main/java/org/sonar/plugins/fsharp/FSharpSonarWayProfile.java]: Ok i will update

**Comment by milbrandt** [comment]
Sonarqube 7.9 LTS is pre-announced for July, thus 8.x will come this year and there the 6.x deprecations will be removed according to their announcements.
Are these bigger issues? Why not fix all the 6.x deprecation in this PR so that everything would be ready for Sonarqube 8?

**Comment by jmecosta** [comment]
> Sonarqube 7.9 LTS is pre-announced for July, thus 8.x will come this year and there the 6.x deprecations will be removed according to their announcements.
> Are these bigger issues? Why not fix all the 6.x deprecation in this PR so that everything would be ready for Sonarqube 8?

time ;) but i pushed the removal in the java side. so SonarQube will be happy. the F# side i still want to keep those one, they might be handy for other stuff

**Comment by milbrandt** [comment]
> time ;) but i pushed the removal in the java side. so SonarQube will be happy. the F# side i still want to keep those one, they might be handy for other stuff

That's a self-evident reason for everyone ;) But IMHO better to fix everything in one release of the plugin. Up to v1.0.5 it's using deprecated API, starting with next release (v1.0.6 maybe) all deprecated API is removed and ready for SonarQube 8.

Seems like you removed alredy 3/4 - I only saw `org.sonar.api.batch.fs.FileSystem.files` still in the FSharpSensor. (And the merge conflict - this shouldn't be hard and I can support you there).

For me it's fine to leave the F# code in for now.

**Comment by jmecosta** [comment]
> > time ;) but i pushed the removal in the java side. so SonarQube will be happy. the F# side i still want to keep those one, they might be handy for other stuff
> 
> That's a self-evident reason for everyone ;) But IMHO better to fix everything in one release of the plugin. Up to v1.0.5 it's using deprecated API, starting with next release (v1.0.6 maybe) all deprecated API is removed and ready for SonarQube 8.
> 
> Seems like you removed alredy 3/4 - I only saw `org.sonar.api.batch.fs.FileSystem.files` still in the FSharpSensor. (And the merge conflict - this shouldn't be hard and I can support you there).
> 
> For me it's fine to leave the F# code in for now.

uhm, ok i still chekc those

**Comment by jmecosta** [comment]
should be ok now, if i didnt miss anything

**Review by milbrandt — APPROVED** [review]

**Comment by milbrandt** [comment]
In `FSharpSensor` line 442 and 469 is it correct to set the value 1 or should that be `value`? This is unchanged in this PR, but Maybe you know it anyway @jmecosta

**Comment by jmecosta** [comment]
> ne 442 and 469

it should be correct, putting more that 1 means one line has more that one comment line. value is the increment of the all file

**Pull request merged**